backend/app/usecase/url/urlretriever_test.go

```diff
@@ -101,3 +102,200 @@ func TestUrlRetriever_GetURL(t *testing.T) {
+			},
+			hasErr: false,
+			expectedURLs: []entity.URL{
+				{
+					Alias:       "google",
+					OriginalURL: "https://www.google.com/",
+				},
+				{
+					Alias:       "short",
+					OriginalURL: "https://github.com/short-d/short/",
+				},
+			},
+		},
+		{
+			name: "user has no URL",
+			urls: urlMap{
+				"google": entity.URL{
+					Alias:       "google",
+					OriginalURL: "https://www.google.com/",
+				},
+				"short": entity.URL{
+					Alias:       "short",
+					OriginalURL: "https://github.com/short-d/short/",
+				},
+				"mozilla": entity.URL{
+					Alias:       "mozilla",
+					OriginalURL: "https://www.mozilla.org/",
+				},
+			},
+			users: []entity.User{
+				{
+					ID:    "12345",
+					Name:  "Test User",
+					Email: "test@gmail.com",
+				}, {
+					ID:    "12345",
+					Name:  "Test User",
+					Email: "test@gmail.com",
+				}, {
+					ID:    "12345",
+					Name:  "Test User",
+					Email: "test@gmail.com",
+				},
+			},
+			createdURLs: []entity.URL{
+				{
+					Alias:       "google",
+					OriginalURL: "https://www.google.com/",
+				},
+				{
+					Alias:       "short",
+					OriginalURL: "https://github.com/short-d/short/",
+				},
+				{
+					Alias:       "mozilla",
+					OriginalURL: "https://www.mozilla.org/",
+				},
+			},
+			user: &entity.User{
+				ID:    "12346",
+				Name:  "Test User 2",
+				Email: "test2@gmail.com",
+			},
+			hasErr:       false,
+			expectedURLs: []entity.URL{},
+		}, {
+			name: "user is nil",
+			urls: urlMap{
+				"google": entity.URL{
+					Alias:       "google",
+					OriginalURL: "https://www.google.com/",
+				},
+				"short": entity.URL{
+					Alias:       "short",
+					OriginalURL: "https://github.com/short-d/short/",
+				},
+				"mozilla": entity.URL{
+					Alias:       "mozilla",
+					OriginalURL: "https://www.mozilla.org/",
+				},
+			},
+			users: []entity.User{
+				{
+					ID:    "12345",
+					Name:  "Test User",
+					Email: "test@gmail.com",
+				}, {
+					ID:    "12345",
+					Name:  "Test User",
+					Email: "test@gmail.com",
+				}, {
+					ID:    "12346",
+					Name:  "Test User 2",
+					Email: "test2@gmail.com",
+				},
+			},
+			createdURLs: []entity.URL{
+				{
+					Alias:       "google",
+					OriginalURL: "https://www.google.com/",
+				},
+				{
+					Alias:       "short",
+					OriginalURL: "https://github.com/short-d/short/",
+				},
+				{
+					Alias:       "mozilla",
+					OriginalURL: "https://www.mozilla.org/",
+				},
+			},
+			user:         nil,
+			hasErr:       true,
+			expectedURLs: []entity.URL{},
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			t.Parallel()
+
+			fakeURLRepo := repository.NewURLFake(testCase.urls)
+			fakeUserURLRelationRepo := repository.NewUserURLRepoFake(testCase.users, testCase.createdURLs)
+			retriever := NewRetrieverPersist(&fakeURLRepo, &fakeUserURLRelationRepo)
+
+			urls, err := retriever.GetURLsByUser(testCase.user)
+			if testCase.hasErr {
```

Comment:
I think it is better to rewrite, something like:

```
mdtest.Equal(t, t.hasErr, err != nil)
mdtest.Equal(t, testCase.expectedURLs, urls)
```

Reply:
I think this introduces extract logical efforts for readers to think.

Reply:
Let's simplify everything for the tests.

Reply:
I recommend keep as it is to hold up the readability of the test.